src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -81,21 +83,77 @@ public static FunctionPointerMethodSymbol CreateMethodFromSource(FunctionPointer
                 callingConvention,
                 refKind,
                 returnType,
+                refReadonlyModifiers,
                 syntax,
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
         }
 
+        public static FunctionPointerMethodSymbol CreateFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
+        {
+            Debug.Assert(retAndParamTypes.Length > 0);
+
+            ParamInfo<TypeSymbol> retInfo = retAndParamTypes[0];
+            var returnType = TypeWithAnnotations.Create(retInfo.Type, customModifiers: CSharpCustomModifier.Convert(retInfo.CustomModifiers));
+            var refCustomMods = CSharpCustomModifier.Convert(retInfo.RefCustomModifiers);
+
+            RefKind refKind = retInfo.IsByRef switch
+            {
+                false => RefKind.None,
+                true when CustomModifierUtils.HasInAttributeModifier(refCustomMods) => RefKind.RefReadOnly,
+                true => RefKind.Ref,
+            };
+
+            return new FunctionPointerMethodSymbol(
+                callingConvention,
+                refKind,
+                refCustomMods,
+                returnType,
+                retAndParamTypes.AsSpan()[1..]);
+        }
+
+        private static ImmutableArray<FunctionPointerParameterSymbol> MakeParametersFromMetadata(ReadOnlySpan<ParamInfo<TypeSymbol>> parameterTypes, FunctionPointerMethodSymbol parent)
```

Comment:
>MakeParametersFromMetadata [](start = 70, length = 26)

It looks like there is only one call site, consider making this a local function